domain/pom.xml

```diff
@@ -96,8 +96,9 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
-                    <source>16</source>
-                    <target>16</target>
+                    <source>17</source>
+                    <target>17</target>
+                  <compilerArgs>--enable-preview</compilerArgs>
```

Comment:
Preview features in production code? For what feature do we require Java's preview?

Reply:
I think this was added by me to try out the new [switch](https://blog.adamgamboa.dev/switch-expression-on-java-17/ ) feature of java 17